frontend/src/components/Recipes/Add/RecipeForm.tsx

```diff
@@ -1,42 +1,184 @@
+    }
+
+    async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+        e.preventDefault()
+
+        const form = new FormData(e.currentTarget)
+
+        const sendData = await fetch(`${endpoint}/recipes`, {
+            method: 'POST',
+            headers: {
+                enctype: 'multipart/form-data',
+            },
+            body: form,
+        })
+
+        if (sendData.ok) {
+            console.log('Recipe added successfully')
+        } else {
+            console.log('Error adding recipe')
+        }
+    }
+
+    function handleReset() {
+        setValues(initialValues)
+        setErrorVisibility(initialErrorVisibility)
+    }
+
     return (
         <>
             <form
                 className="flex flex-col gap-4"
-                action={`${endpoint}/recipes`}
-                method="POST"
-                encType="multipart/form-data"
+                onSubmit={handleSubmit}
+                onReset={handleReset}
             >
-                <div className="[&>*]:block">
-                    <label htmlFor="name">Name</label>
-                    <input type="text" id="name" name="name" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="ingredients">Ingredients</label>
-                    <input type="text" id="ingredients" name="ingredients" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="instructions">Instructions</label>
-                    <textarea id="instructions" name="instructions" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="cuisine">Cuisine</label>
-                    <input type="number" id="cuisine" name="cuisineId" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="dietary">Dietary</label>
-                    <input type="number" id="dietary" name="dietaryId" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="difficulty">Difficulty</label>
-                    <input type="number" id="difficulty" name="difficultyId" />
-                </div>
-                <div className="[&>*]:block">
-                    <label htmlFor="image">Image</label>
-                    <input type="file" id="image" name="image" />
-                </div>
+                {formInputs.map((input) => {
+                    const { errorMessage, ...inputProps } = input
+
+                    return (
+                        <div className="[&>label]:block" key={input.id}>
+                            <label htmlFor={input.name}>{input.label}</label>
+                            <input
+                                onChange={handleChange}
+                                value={
+                                    values[input.name as keyof typeof values]
```

Comment:
Here it can turn out to be very useful to type formInputs in a more precise way to avoid the explicit cast that you currently need when accessing a value. This is an example in which letting typescript infer the type lets you lose a stricter constraint on the name. I would play a bit on type formInputs in a more precise way.

Aim: removing `as keyof typeof values` when you access a value